inspirehep/modules/records/serializers/fields_export.py

```diff
@@ -0,0 +1,318 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Instructions on where to find relevant bits of data in API."""
+
+from __future__ import absolute_import, division, print_function
+from idutils import is_arxiv_post_2007
+from inspire_utils.record import get_value
+from inspire_utils.date import PartialDate
+from inspirehep.modules.hal.utils import get_page_artid, get_conference_record, get_conference_title
+from babel import Locale
+from isbn_hyphenate import hyphenate, IsbnError
+
+
+MAX_AUTHORS_BEFORE_ET_AL = 10  # According to CSE stylebook
+
+# Fields shared among all bibtex entries:
+COMMON_FIELDS_FOR_ENTRIES = ['key', 'SLACcitation', 'archivePrefix', 'doi', 'eprint', 'month', 'note', 'primaryClass',
+                             'title', 'url', 'year']
+
+# Fields for a given bibtex entry. Since we're trying to match as many as possible
+# it doesn't matter whether they're mandatory or optional
+FIELDS_FOR_ENTRY_TYPE = {
+    'techreport': ['author', 'collaboration', 'number', 'address', 'type', 'institution'],
+    'phdthesis': ['reportNumber', 'school', 'address', 'type', 'author'],
+    'inproceedings': ['publisher', 'author', 'series', 'booktitle', 'number', 'volume', 'reportNumber', 'editor',
+                      'address', 'organization', 'pages'],
+    'misc': ['howpublished', 'reportNumber', 'author'],
+    'mastersthesis': ['reportNumber', 'school', 'address', 'type', 'author'],
+    'proceedings': ['publisher', 'series', 'number', 'volume', 'reportNumber', 'editor', 'address', 'organization',
+                    'pages'],
+    'book': ['publisher', 'isbn', 'author', 'series', 'number', 'volume', 'edition', 'editor', 'reportNumber',
+             'address'],
+    'inbook': ['chapter', 'publisher', 'author', 'series', 'number', 'volume', 'edition', 'editor', 'reportNumber',
+               'address', 'type', 'pages'],
+    'article': ['author', 'journal', 'collaboration', 'number', 'volume', 'reportNumber', 'pages'],
+}
+
+
+def make_extractor():
+    """
+    Returns:
+        function: a decorator with a store for pre-processing/extracting functions.
+    """
+    store = {}
+
+    def extractor(field):
+        def decorator(decorated_function):
+            store[field] = decorated_function
+            return decorated_function
+        return decorator
+    extractor.store = store
+    return extractor
+
+
+def bibtex_document_type(doc_type, obj):
+    """
+    Args:
+        doc_type (text_type): INSPIRE document type.
+        obj (dict): literature record.
+
+    Returns:
+        text_type: bibtex document type for the given INSPIRE entry.
+    """
```

Comment:
PEP257 talks about the conventions of the Python community with regards to docstrings. Here's how to rewrite this one so that it conforms to them (https://www.python.org/dev/peps/pep-0257/):
```python
"""Return the BibTeX entry type.

Maps the INSPIRE ``document_type`` to a BibTeX entry type. Also checks
``thesis_info.degree_type`` in case it's a thesis, as it stores the information
on which kind of thesis we're dealing with.
"""
```